### DISCUSSION TITLE
General Discussion

### DISCUSSION THREAD

**Discussion by andrew-codechimp** [andrew-codechimp/bn-device-test#27] [General]
I've disabled the actions that validate the integration, not needed for testing the actions and just slow things down.

**Comment by bmos** [comment]
Can you enable issues?

I want to test:
1. Create a new device PR via an issue
2. Create a manual new device PR via a fork
3. Create a second new device PR via an issue

I think this should be effective in confirming that the auto-sorting has resolved the merge commit issues that happened with the issue template.

**Reply by andrew-codechimp** [reply]
Just enabled, totally forgot we were testing an issue template lol

**Comment by bmos** [comment]
Okay, I feel like it might be tested.

1. I suggested ABC1 via [an issue](https://github.com/andrew-codechimp/bn-device-test/issues/37) and [a PR](https://github.com/andrew-codechimp/bn-device-test/pull/38) was automatically created. No sorting occurred.
2. I suggested ABC2 via [a manual PR](https://github.com/andrew-codechimp/bn-device-test/pull/39) from [a branch](https://github.com/bmos/bn-device-test/tree/patch-1) on my fork. No sorting occurred.
3. I suggested ABC4 via [a manual PR](https://github.com/andrew-codechimp/bn-device-test/pull/42) from [a branch](https://github.com/bmos/bn-device-test/tree/patch-2) on my fork. No sorting occurred.
4. I merged ABC4. The entire library was [sorted automatically](https://github.com/andrew-codechimp/bn-device-test/commit/abb22f3a5cb1cf64b22170a1b044ce5bbd5ed5fe).
5. ABC1 and ABC2 PRs are both ready to merge with no conflicts to resolve.

**Comment by bmos** [comment]
There is one circumstance where sorting could occur during a PR.
If the person submitting the manual PR has Actions enabled on their fork, then their PR will be sorted (by the action running on their repo) after their commit and again (by the action running on your repo) after it's merged.

**Reply by andrew-codechimp** [reply]
I think that's an edge case that we could handle in the readme for manual submissions.

**Comment by andrew-codechimp** [comment]
I see you keep trying to use the ScratchAddons/validate-json-action.
This looks more modern and doesn't throw the warning but whatever I've tried it comes back as success but if you look at it's output it's saying there's no json to validate so something's a bit odd with this action.

I see GitHub have postponed deprecating the output used in the older validator, apparently there are too many actions using this method for them to action it yet so we have some time.

**Reply by bmos** [reply]
Oh awesome, I hadn't noticed that deadline was extended.

**Comment by andrew-codechimp** [comment]
Just added auto closing of the issue with a Thanks message after the PR is created, will save me having to do two activities per device.

**Reply by bmos** [reply]
Awesome, I was wondering how best to handle that.
If "closes #2" or something similar is in the comment on the merge message it would auto-close but I agree closing it once the PR exists makes the most sense/is easiest.

**Comment by bmos** [comment]
I thought of two things to check:
1. We should test what happens if people try to submit a definition that already exists.
2. We should check what happens when people use weird device names with punctuation.

**Comment by bmos** [comment]
Doing a combined test in #59
Looks like the issue is closed and the PR is not created.
That's good, but the message in the Issue is misleading/not transparent.

**Reply by bmos** [reply]
no issues with underlines in device manufacturer:
https://github.com/andrew-codechimp/bn-device-test/actions/runs/7359796509/job/20035087168#step:10:48

doesn't create the new pr because it's not ahead of main:
https://github.com/andrew-codechimp/bn-device-test/actions/runs/7359796509/job/20035087168#step:10:77

**Reply by bmos** [reply]
I think the easiest solution is to just change the message slightly:
> Thanks for the contribution. We're auto-closing this issue. If it's a new device, a pull request will be created that will be reviewed and merged.

**Comment by andrew-codechimp** [comment]
You'll see I've added a new doc gen capability to the sort workflow so I have a nice easy to read markdown file of the library as well.
I've never been great at thorough testing but think with the efforts you've put in recently we're ready for a release.
What do you think, anything you still want to check?

Do you want to update your PR or scrap that and I'll manually commit?
